### PR TITLE
test: 11136 - Create hapi tests for selfDestruct operation with system accounts as beneficiaries

### LINKED ISSUE #11136: System accounts disambiguation: Create hapi tests for selfDestruct operation for system accounts
### Problem

Calling operations on system accounts in hedera may differ from ethereum.
We need to add hapi tests for them and check and document the behaviour.

### Solution

Add tests for **selfDestruct** operation involving the system accounts and ensure that these tests conform to the following expected behavior.
image

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:
Calling operations on system accounts in hedera may differ from ethereum.
We need to add hapi tests for them and check and document the behaviour.

This PR adds tests for `selfDestruct` operation involving the system accounts and ensure that these tests conform to the following expected behavior:

image

**Related issue(s)**:

Fixes #11136

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/evm/Evm46ValidationSuite.java [hunk 1]
```diff
@@ -1516,16 +1523,81 @@ contract, BALANCE_OF, mirrorAddrWith(systemAccounts.get(i)))
...
+        final var opsArray = new HapiSpecOperation[nonExistingAccountsOps.length + existingAccountsOps.length];
+
+        System.arraycopy(nonExistingAccountsOps, 0, opsArray, 0, nonExistingAccountsOps.length);
+        System.arraycopy(existingAccountsOps, 0, opsArray, nonExistingAccountsOps.length, existingAccountsOps.length);
+
+        return defaultHapiSpec("testSelfDestructForSystemAccountsWithSigRequiredTrue")
+                .given(
+                        cryptoCreate(BENEFICIARY)
+                                .balance(ONE_HUNDRED_HBARS)
+                                .receiverSigRequired(true)
```
Comment (0xivanov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/evm/Evm46ValidationSuite.java:1583]: Some of the ops should probably fail here, need to sync with @stoqnkpL about this, as I am not 100% sure.

Comment (0xivanov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/evm/Evm46ValidationSuite.java]: Turns out all the system accounts that exist (800-999 from what i've seen) have `receiverSigRequired=false` so the transaction would always succeed.

**Review by tinker-michaelj — APPROVED** [review]
Isn't this actually testing whether system accounts can be used as _beneficiaries_ of `selfdestruct`?

Assuming that's the goal, LGTM!

**Comment by 0xivanov** [comment]
> Isn't this actually testing whether system accounts can be used as _beneficiaries_ of `selfdestruct`?
> 
> Assuming that's the goal, LGTM!
Yes, this is the goal :) . The PR title might be misleading, will modify it.

**Pull request merged**